src/components/OverflowMenu/OverflowMenu.tsx

```diff
@@ -0,0 +1,66 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import React, { useState } from 'react';
+import { usePopper } from 'react-popper';
+import { IconDotsHorizontal } from '@foundation/Icon';
+import { Button, ButtonRounding, ButtonSize, ButtonStyle, ButtonType } from '..';
+import { OverflowMenuItem, OverflowMenuItemProps } from './OverflowMenuItem';
+
+export interface OverflowMenuProps {
+    items: OverflowMenuItemProps[];
+}
+
+export const OverflowMenu = ({ items }: OverflowMenuProps) => {
+    const [isMenuOpened, setIsMenuOpened] = useState(false);
+    const [menuContainerRef, setMenuContainerRef] = useState<HTMLElement | null>(null);
```

Comment:
Here for the ref you can instead use `useRef`:
`const menuContainerRef = useRef<HTMLButtonElement | null>(null);`

Reply:
As discussed on our call, I'm using the popper.js recommended hook for updating the container properly when the element changes. The docs link: https://popper.js.org/react-popper/v2/